scripts/install_third_party.py

```diff
@@ -172,50 +216,50 @@ def download_manifest_files(source_url):
     """
     manifest_data = return_json(source_url)
     dependencies = manifest_data['dependencies']
-    for data in dependencies:
-        dependency = dependencies[data]
-        for dependency_id in dependency:
-            dependency_contents = dependency[dependency_id]
-            if 'srcUrl' in dependency_contents:
-                DEPENDENCY_REV = dependency_contents['version']
-                DEPENDENCY_URL = dependency_contents['srcUrl']
-                DEPENDENCY_FILES = dependency_contents['files']
-                TARGET_DIRNAME = (
-                    dependency_contents['targetDirPrefix'] + DEPENDENCY_REV)
-                DEPENDENCY_DST = os.path.join(
-                    TARGET_DOWNLOAD_DIRS[data], TARGET_DIRNAME)
-                download_files(DEPENDENCY_URL, DEPENDENCY_DST, DEPENDENCY_FILES)
-
-            elif 'zipUrl' in dependency_contents:
-                DEPENDENCY_REV = dependency_contents['version']
-                DEPENDENCY_URL = dependency_contents['zipUrl']
-                if 'rootDir' in dependency_contents:
-                    DEPENDENCY_ZIP_ROOT_NAME = dependency_contents['rootDir']
-                else:
-                    DEPENDENCY_ZIP_ROOT_NAME = (
-                        dependency_contents['rootDirPrefix'] + DEPENDENCY_REV)
-
-                if 'targetDir' in dependency_contents:
-                    DEPENDENCY_TARGET_ROOT_NAME = (
-                        dependency_contents['targetDir'])
-                else:
+    for data, dependency in dependencies.items():
```

Comment:
I think it's better to do just one call to validate_manifest() at the outset, before downloading anything at all. That way you don't have to deal with cases where some of the stuff is downloaded and some isn't.